helix-tui/src/backend/crossterm.rs

```diff
@@ -154,7 +159,12 @@ where
 
     fn restore(&mut self, config: Config) -> io::Result<()> {
         // reset cursor shape
-        write!(self.buffer, "\x1B[0 q")?;
+        let cursor_normal = self
+            .capabilities
+            .cursor_normal
+            .as_deref()
+            .unwrap_or("\x1B[0 q");
+        write!(self.buffer, "{cursor_normal}")?;
```

Comment:
```suggestion
        self.buffer.write_all(cursor_normal.as_bytes())?;
```

here the formatting is just overhead, we could convert to a byte slice instead directly and avoid all that